### PR TITLE
Add link to open support ticket from Altis menu

### LINKED ISSUE #132: Add link to open support ticket from Altis menu
Alongside #126, we should also link directly to the support ticket page.

The format for the link should be `https://dashboard.altis-dxp.com/#/support/new?applications[]={stack-name}` if we have the environment name, or `https://dashboard.altis-dxp.com/#/support/new` otherwise.

### PR DESCRIPTION
**Pull request by dashaluna**
Fixes #132 

@roborourke let me know if it's ok to keep in within the env indicator code? Otherwise, I'll move it into a separate folder (let me know the name of it).

### CODE REVIEW

File: inc/environment_indicator/namespace.php [hunk 1]
```diff
@@ -88,6 +89,31 @@ function add_admin_bar_dashboard_link( WP_Admin_Bar $wp_admin_bar ) {
...
+ *
+ * @param WP_Admin_Bar $wp_admin_bar WP_Admin_Bar instance, passed by reference.
+ */
+function add_admin_bar_support_ticket_link( WP_Admin_Bar $wp_admin_bar ) {
+	if ( ! is_admin_bar_showing() ) {
+		return;
+	}
+
+	$env_name    = get_environment_name();
+	$support_url = 'unknown' !== $env_name ? "https://dashboard.altis-dxp.com/#/support/new?applications[]={$env_name}" : 'https://dashboard.altis-dxp.com/#/support/new';
```
Comment (rmccue) [inline inc/environment_indicator/namespace.php]: This line is super long; can we move the base URL (`.../support/new`) to a constant?

Also, this should use either `sprintf` or `add_query_arg` for the argument, and it needs to be URL encoded.

Comment (dashaluna) [reply inc/environment_indicator/namespace.php]: @rmccue thanks.

`add_query_arg` adds the args before the `#` in the URL, so getting `https://dashboard.altis-dxp.com/?applications%5B%5D=test#/support/new`, I guess anchor comes after the args in the URL structure.

As for `sprintf` I remember a feedback that URL should be escaped in full, not parts. I've updated code. Let me know if that works.

File: inc/environment_indicator/namespace.php [hunk 2]
```diff
@@ -88,6 +89,32 @@ function add_admin_bar_dashboard_link( WP_Admin_Bar $wp_admin_bar ) {
...
+ * @param WP_Admin_Bar $wp_admin_bar WP_Admin_Bar instance, passed by reference.
+ */
+function add_admin_bar_support_ticket_link( WP_Admin_Bar $wp_admin_bar ) {
+	if ( ! is_admin_bar_showing() ) {
+		return;
+	}
+
+	$env_name    = get_environment_name();
+	$support_url = 'https://dashboard.altis-dxp.com/#/support/new';
+	$support_url = 'unknown' === $env_name ? $support_url : "{$support_url}?applications[]={$env_name}";
```
Comment (roborourke) [inline inc/environment_indicator/namespace.php]: Ternary operators are a bit harder to read in general. In this instance you're not actually modifying the URL when the condition is true so a basic `if` statement would be a better fit and easier to understand:
```suggestion
	if ( $env_name !== 'unknown' ) {
		$support_url .= sprintf( '?applications[]=%s', urlencode( $env_name ) );
	}
```

File: inc/environment_indicator/namespace.php [hunk 3]
```diff
@@ -88,6 +89,32 @@ function add_admin_bar_dashboard_link( WP_Admin_Bar $wp_admin_bar ) {
...
+ *
+ * @param WP_Admin_Bar $wp_admin_bar WP_Admin_Bar instance, passed by reference.
+ */
+function add_admin_bar_support_ticket_link( WP_Admin_Bar $wp_admin_bar ) {
+	if ( ! is_admin_bar_showing() ) {
+		return;
+	}
+
+	$env_name    = get_environment_name();
+	$support_url = 'https://dashboard.altis-dxp.com/#/support/new';
```
Comment (rmccue) [inline inc/environment_indicator/namespace.php]: This should be a `const` in the namespace rather than a variable here

Comment (roborourke) [reply inc/environment_indicator/namespace.php]: Think it's helpful to explain why you'd do that as well

**Comment by dashaluna** [comment]
Thank you both for feedback and patience(!)
Hopefully refactored code is ok :)

**Pull request merged**